libnd4j/include/helpers/ConstantShapeHelper.h

```diff
@@ -38,12 +38,25 @@ class SD_LIB_EXPORT ConstantShapeHelper {
  private:
   std::mutex _mutex;
   std::vector<SD_MAP_IMPL<ShapeDescriptor, ConstantShapeBuffer>> _cache;
-
+#if defined(__NEC__)
+  bool _cache_existing_pointers = true;
+#endif
   ConstantShapeHelper();
 
  public:
   ~ConstantShapeHelper() = default;
 
+#if defined(__NEC__)
+  //Warning: Use it with caution. please, restore it to the previous state to avoid interfering internals
```

Comment:
A bit more context around what this is for and why it's good/bad would  be nice. That will affect future decisions when we read the code again.